src/Credentials/ServiceAccountCredentials.php

```diff
@@ -75,11 +75,13 @@ class ServiceAccountCredentials extends CredentialsLoader implements SignBlobInt
      *   as an associative array
      * @param string $sub an email address account to impersonate, in situations when
      *   the service account has been delegated domain wide access.
+     * @param string $targetAudience The audience for the ID token.
      */
     public function __construct(
         $scope,
         $jsonKey,
-        $sub = null
+        $sub = null,
+        $targetAudience = null
```

Comment:
In `GCECredentials::__construct()`, we throw if `$scope` and `$targetAudience` are both supplied, but here we don't. Is that because `$scope` isn't optional in the signature? If there aren't other reasons, should we just tell users to set `$scope` to `null` if they provide `$targetAudience`? Nasty but probably clearer and less prone to accidentally misusing the method and getting something unexpected.

Reply:
When both scope and target audience are set, the OAuth2 Token endpoint returns the following 400 error:

```json
{
  "error": "invalid_request",
  "error_description": "Cannot specify both scope and target audience."
}
```

So really, it's just an oversight and I can add the exception here as well!